helfi_tpr.views_execution.inc

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/**
+ * @file
+ * Views alter for helfi_tpr.
+ */
+
+use Drupal\views\Plugin\views\query\QueryPluginBase;
+use Drupal\views\ViewExecutable;
+
+/**
+ * Implements hook_views_query_alter().
+ */
+function helfi_tpr_views_query_alter(ViewExecutable $view, QueryPluginBase $query) {
+  if ($view->id() == 'unit_search') {
+    // Use the CASE function from helfi_tpr_views_pre_execute() for sorting.
+    $query->orderby[0]['field'] = 'name_sort';
+    $query->orderby[0]['direction'] = 'ASC';
+  }
+}
+
+/**
+ * Implements hook_views_pre_execute().
+ */
+function helfi_tpr_views_pre_execute(ViewExecutable $view) {
+  if ($view->id() == 'unit_search') {
+    // CASE function for using the name_override field as default for sorting.
+    // If that field is empty, use the name field.
+    $view->build_info['query']->addExpression('CASE WHEN tpr_unit_field_data.name_override IS NULL THEN tpr_unit_field_data.name ELSE tpr_unit_field_data.name_override END', 'name_sort');
```

Comment:
Should this use curly brackets with table names and square brackets with column names? See e.g. https://www.drupal.org/node/2986894 and https://www.drupal.org/docs/develop/standards/sql-coding-conventions#reserved-words-drupal-8

Reply:
Doesn't seem to apply to expressions, see: https://www.drupal.org/docs/7/api/database-api/dynamic-queries/expressions

It seems to work, though, even if it looks a bit funky, like `{table_name}.[column_name]`.